src/morale.cpp

```diff
@@ -476,39 +556,42 @@ void player_morale::on_effect_int_change( const efftype_id &eid, int intensity,
     if( eid == effect_took_prozac && bp == num_bp ) {
         set_prozac( intensity != 0 );
     } else if( eid == effect_cold && bp < num_bp ) {
-        cold[bp] = intensity;
+        body_parts[bp].cold = intensity;
     } else if( eid == effect_hot && bp < num_bp ) {
-        hot[bp] = intensity;
+        body_parts[bp].hot = intensity;
     }
 }
 
 void player_morale::set_worn( const item &it, bool worn )
 {
     const bool just_fancy = it.has_flag( "FANCY" );
     const bool super_fancy = it.has_flag( "SUPER_FANCY" );
-
-    if( just_fancy || super_fancy ) {
-        const int sign = ( worn ) ? 1 : -1;
-
-        for( int i = 0; i < num_bp; i++ ) {
-            const auto bp = static_cast<body_part>( i );
-            if( it.covers( bp ) ) {
-                covered[i] = std::max( covered[i] + sign, 0 );
+    const bool anyhow_fancy = just_fancy || super_fancy;
+    const int sign = ( worn ) ? 1 : -1;
+
+    for( int i = 0; i < num_bp; ++i ) {
+        if( it.covers( static_cast<body_part>( i ) ) ) {
+            body_parts[i].mod_covered( sign );
+            if( anyhow_fancy ) {
+                body_parts[i].mod_covered_fancy( sign );
             }
         }
+    }
 
-        if( super_fancy ) {
-            super_fancy_bonus += 2 * sign;
-        }
-
+    if( super_fancy ) {
+        super_fancy_bonus = std::max( super_fancy_bonus + 2 * sign, 0 );
```

Comment:
> This super fancy bonus stacks regardless of body parts. 

Same here. It's the original behavior which I didn't change. If it needs to be changed, I'd prefer not mix it up with the refactoring in one PR.